packages/instantsearch-core/src/widgets/index-widget.ts

```diff
@@ -686,10 +682,8 @@ export const index = (widgetParams: IndexWidgetParams): IndexWidget => {
 
       localWidgets.forEach((widget) => {
         warning(
-          // if it has NO getWidgetState or if it has getWidgetUiState, we don't warn
-          // aka we warn if there's _only_ getWidgetState
-          !widget.getWidgetState || Boolean(widget.getWidgetUiState),
-          'The `getWidgetState` method is renamed `getWidgetUiState` and will no longer exist under that name in InstantSearch.js 5.x. Please use `getWidgetUiState` instead.'
+          !(widget as any).getWidgetState,
+          'The `getWidgetState` method is renamed `getWidgetUiState` and no longer exists under that name in InstantSearch.js 5.x. Please use `getWidgetUiState` instead.'
```

Comment:
Nitpick: the only scenario I see where this would be triggered is for custom widgets created more than 4 years ago (the point from which the removed warning got added). We could go one step further and remove this, wdyt?

Reply:
true, I didn't realise the warning was this old. Will remove it too